dbhydra/dbhydra_core.py

```diff
@@ -1042,124 +1035,136 @@ def __init__(self,name="new_db",config_file="config.ini"):
+                value = int(value)  # integers
             except:
-                value=value.split("'")[1] #strings
-            return(variable,value)
-        
-        variable,value=split_assign(variable_assign)
-        df=self.select_to_df()
+                value = value.split("'")[1]  # strings
+            return (variable, value)
+
+        variable, value = split_assign(variable_assign)
+        df = self.select_to_df()
         print(where)
-        print(variable,value)
-        if where is None:    
-            df[variable]=value  
+        print(variable, value)
+        if where is None:
+            df[variable] = value
             print(df)
-        else:    
-            where_variable,where_value=split_assign(where)
-            df[variable] = df[where_variable].apply(lambda x: value if x==where_value else x)
+        else:
+            where_variable, where_value = split_assign(where)
+            df[variable] = df[where_variable].apply(lambda x: value if x == where_value else x)
         self.replace_from_df(df)
 
-    def delete(self,where=None):    
+    def delete(self, where=None):
         def split_assign(variable_assign):
-            variable=variable_assign.split("=")[0]
-            value=variable_assign.split("=")[1]
+            variable = variable_assign.split("=")[0]
+            value = variable_assign.split("=")[1]
             try:
-                value=int(value) #integers
+                value = int(value)  # integers
             except:
-                value=value.split("'")[1] #strings
-            return(variable,value)
-       
-        df=self.select_to_df()
-        if where is None:    
-            df=df.iloc[0:0]
+                value = value.split("'")[1]  # strings
+            return (variable, value)
+
+        df = self.select_to_df()
+        if where is None:
+            df = df.iloc[0:0]
             print(df)
-        else:    
-            where_variable,where_value=split_assign(where)
-            df.drop(df[df[where_variable]==where_value].index, inplace=True)
+        else:
+            where_variable, where_value = split_assign(where)
+            df.drop(df[df[where_variable] == where_value].index, inplace=True)
         self.replace_from_df(df)
-        
-#dataframe - dictionary auxiliary functions     
-def df_to_dict(df,column1,column2):
-    dictionary=df.set_index(column1).to_dict()[column2]
-    return(dictionary)
-
-def dict_to_df(dictionary,column1,column2):
-    df=pd.DataFrame(list(dictionary.items()), columns=[column1, column2])
-    return(df)
+
+
+# dataframe - dictionary auxiliary functions
+def df_to_dict(df, column1, column2):
+    dictionary = df.set_index(column1).to_dict()[column2]
+    return (dictionary)
+
+
+def dict_to_df(dictionary, column1, column2):
+    df = pd.DataFrame(list(dictionary.items()), columns=[column1, column2])
+    return (df)
+
+
+from contextlib import contextmanager
+
+
+@contextmanager
```

Comment:
Wouldn't this be better as a natural part of every "DB" object? so put for example in AbstractDb?

Reply:
good point, dont know why i did not do that in the first place
moved to abstract db